plugins/modules/collaborator_information.py

```diff
@@ -0,0 +1,351 @@
+            # collab_output['gravatar_id'] = collaborator.gravatar_id
+            # collab_output['url'] = collaborator.url
+            # collab_output['html_url'] = collaborator.html_url
+            # collab_output['followers_url'] = collaborator.followers_url
+            # collab_output['following_url'] = collaborator.following_url
+            # collab_output['gists_url'] = collaborator.gists_url
+            # collab_output['starred_url'] = collaborator.starred_url
+            # collab_output['subscriptions_url'] = collaborator.subscriptions_url
+            # collab_output['organizations_url'] = collaborator.organizations_url
+            # collab_output['repos_url'] = collaborator.repos_url
+            # collab_output['events_url'] = collaborator.events_url
+            # collab_output['received_events_url'] = collaborator.received_events_url
+            collab_output['type'] = collaborator.type
+            collab_output['site_admin'] = collaborator.site_admin
+            permissions = {
+                'triage': collaborator.permissions.triage,
+                'push': collaborator.permissions.push,
+                'pull': collaborator.permissions.pull,
+                'admin': collaborator.permissions.admin
+                }
+            collab_output['permissions'] = permissions
+
+
+            dict_repo.append(collab_output.copy())
+
+        output[repo] = dict_repo.copy()
+
+    return output
+
+
+def run_module():
+    changed = True
+    module_args = dict(
+        token=dict(type='str', default='John Doe'),
+        organization_name=dict(type='str', default='default'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False,
+        fact=''
+    )
+
+    valid_permissions = ["push", "pull", "admin"]
+
+    # token usage retrieved from module's variables from playbook
+    if(module.params['enterprise_url'] == ''):
+        g = Github(module.params['token'])
+    else:
+        g = Github(module.params['token'], base_url=module.params['enterprise_url'])
+
+    if len(module.params['repos']):
+        for i in range(len(module.params['repos'])):
+            module.params['repos'][i] = module.params['organization_name'] + "/" + module.params['repos'][i]
+
+    current_collaborators = get_collaborators(g,  module.params['repos'])
+
+    if(module.params['collaborators_to_add']):
+        for permission in module.params['collaborators_to_add']:
+            if module.params['collaborators_to_add'][permission].lower() not in valid_permissions:
+                module.exit_json(changed=False, failed=True, msg="Invalid permission: " + module.params['collaborators_to_add'][permission] + ". Permissions must be 'push' 'pull' or 'admin'")
+
+        if len(module.params['collaborators_to_add']) and len(module.params['repos']):
+            add_collaborators(g, module.params['repos'], module.params['collaborators_to_add'])
```

Comment:
Does this code work?  add_collaborators is defined as taking 4 argument.  Here it is being called with 3 arguments.